eva_sub_cli/validators/docker_validator.py

```diff
@@ -135,50 +137,47 @@ def verify_image_available_locally(self):
             logger.debug(f"Container ({container_image}) image is not available locally")
             return False
 
-    def run_container(self):
-        logger.debug(f"Trying to run container {self.container_name}")
-        try:
-            self._run_quiet_command(
-                "Try running container",
-                f"{self.docker_path} run -it --rm -d --name {self.container_name} {container_image}:{container_tag}"
-            )
-            # stopping execution to give some time to container to get up and running
-            time.sleep(5)
-            if not self.verify_container_is_running():
+    def run_container_if_required(self):
+        if self.verify_container_is_running():
+            raise RuntimeError(f"Container ({self.container_name}) is already running. "
+                               f"Did you start multiple validation for the same directory ?")
+        if self.verify_container_is_stopped():
+            logger.warn(f"Container {self.container_name} was stopped but not cleaned up before.")
+            self.try_restarting_container()
+        else:
+            logger.debug(f"Trying to start container {self.container_name}")
+            try:
+                self._run_quiet_command(
+                    "Running container",
+                    f"{self.docker_path} run -it --rm -d --name {self.container_name} {container_image}:{container_tag}"
```

Comment:
BTW the `--rm` flag removes the container after stopping, so that's why we don't need to remove it explicitly...